src/reducers/MapReducer.js

```diff
@@ -47,10 +54,71 @@ const INITIAL_STATE = {
     },
     mapLoaded: false,
     viewStyle: map.SearchOptions.MAP,
-    sites: []
+    sites: [],
+    displaySites: [],
+    filterCriteriaKeys: {accessibility: [], facilities: [], price: []}
+};
+
+const updateFilterKeys = ({filterCriteriaKeys}, filterKey) => {
+    let filterCriteriaSubKey = null;
+
+    if (_.includes(_.keys(accessibility_options), filterKey)) {
+        filterCriteriaSubKey = 'accessibility';
+    } else if (_.includes(_.keys(facilities_options), filterKey)) {
+        filterCriteriaSubKey = 'facilities';
+    } else if (_.includes(_.keys(price_options), filterKey)) {
+        filterCriteriaSubKey = 'price';
+    }
+
+    const keyIsAlreadyInList = _.includes(filterCriteriaKeys[filterCriteriaSubKey], filterKey);
+    let updatedFilterKeyList = _.cloneDeep(filterCriteriaKeys);
+
+    if (keyIsAlreadyInList) {
+        updatedFilterKeyList[filterCriteriaSubKey] = _.reject(filterCriteriaKeys[filterCriteriaSubKey], existingFilterKey => existingFilterKey === filterKey);
+    } else {
+        updatedFilterKeyList[filterCriteriaSubKey] = _.concat(filterCriteriaKeys[filterCriteriaSubKey], filterKey);
+    }
+
+    return updatedFilterKeyList;
 };
 
+const filterSites = ({sites}, updatedFilterKeys) => {
+    if (updatedFilterKeys.accessibility.length === 0 && updatedFilterKeys.facilities.length === 0 && updatedFilterKeys.price.length === 0) {
+        return sites;
+    }
 
+    const filteredSites = _.filter(sites, site => {
+        const {accessibility, price, facilities} = site;
+        let accessibilityMatch = false;
+        let facilitiesMatch = false;
+        let priceMatch = false;
+
+        if (updatedFilterKeys.accessibility.length === 0) {
+            accessibilityMatch = true;
+        } else {
+            const accessibilityFullTextArray = _.map(updatedFilterKeys.accessibility, filterKey => accessibility_options[filterKey]);
+            accessibilityMatch = _.includes(accessibilityFullTextArray, accessibility);
+        }
+
+        if (updatedFilterKeys.facilities.length === 0) {
+            facilitiesMatch = true;
+        } else {
+            const facilitiesFullTextArray = _.map(updatedFilterKeys.facilities, filterKey => facilities_options[filterKey]);
+            facilitiesMatch = _.includes(facilitiesFullTextArray, facilities);
+        }
+
+        if (updatedFilterKeys.price.length === 0) {
+            priceMatch = true;
+        } else {
+            const pricesFullTextArray = _.map(updatedFilterKeys.price, filterKey => price_options[filterKey]);
+            priceMatch = _.includes(pricesFullTextArray, price);
+        }
+
+        return accessibilityMatch && facilitiesMatch && priceMatch;
```

Comment:
The basic logic above is: if a given sub key of the `filterCriteriaKeys` (such as `price` or `facilities`) has no keys to filter by, then we can just say we have a match there. If there *are* keys in the sub key's list, then we need to see if our site matches any of the keys in that sub key list. Finally, we return to see if our site matched on all three things: `facilities`, `accessibility`, and `price`.